### PR TITLE
add pin/unpin to json api

### PR DESCRIPTION
**Pull request by joshblum**

### CODE REVIEW

File: go/chat/server.go [hunk 1]
```diff
@@ -2727,10 +2727,29 @@ func (h *Server) PinMessage(ctx context.Context, arg chat1.PinMessageArg) (res c
 	ctx = globals.ChatCtx(ctx, h.G(), keybase1.TLFIdentifyBehavior_CHAT_GUI, &identBreaks, h.identNotifier)
 	defer h.Trace(ctx, func() error { return err }, "PinMessage")()
 	defer func() { h.setResultRateLimit(ctx, &res) }()
-	_, err = utils.AssertLoggedInUID(ctx, h.G())
+	uid, err := utils.AssertLoggedInUID(ctx, h.G())
```
Comment (mmaxim) [inline go/chat/server.go]: Maybe we should just do this check on the server.

Comment (joshblum) [reply go/chat/server.go]: can't since the id of the msg we're pinning is in the body

Comment (joshblum) [reply go/chat/server.go]: i saw this more of a nicety of trying to prevent accidental pins than a strict enforcement

Comment (mmaxim) [reply go/chat/server.go]: Could include on supersedes, but sounds good.

**Pull request merged**